kolibri/content/api.py

```diff
@@ -24,6 +26,28 @@ class ChannelMetadataCacheViewSet(viewsets.ModelViewSet):
     def get_queryset(self):
         return models.ChannelMetadataCache.objects.all()
 
+    def destroy(self, request, pk=None):
+        '''
+        Destroys the ChannelMetadata object and also the sqlite3 files on the filesystem.
+        '''
+        super(ChannelMetadataCacheViewSet, self).destroy(request)
+
+        if self.delete_content_db_file(pk):
+            response_msg = 'Channel {} removed from device'.format(pk)
+        else:
+            response_msg = 'Channel {} removed, but no content database was found'.format(pk)
+
+        return Response(response_msg)
+
+    def delete_content_db_file(self, channel_id):
+        db_path = get_content_database_file_path(channel_id)
+
+        if os.path.isfile(db_path):
+            os.remove(db_path)
+            return True
+        else:
+            return False
```

Comment:
Can be more Pythonic, and also avoid a race condition (in case the file is deleted in parallel between lines 45 and 46), to use try/except:

```
try:
    os.remove(db_path)
    return True
except OSError, e:
    return False
```